### PR TITLE
Point of Contact - Phase 1

### LINKED ISSUE #416: [Filing] Integrate existing `Point of Contact` form - Phase 1 - just get it in there
https://github.com/cfpb/sbl-frontend/blob/main/src/pages/PointOfContact/index.tsx

### PR DESCRIPTION
**Pull request by meissadia**
Closes #416 

## Changes

- Integrates form into workflow
- Enables saving POC info to API
- Populates form with existing POC contact when available
- Only saves data to API if it has changed

## Screenshots
What we see:
- User populates form data
- Form validates content and displays an error for missing/malformed fields
- On initial successful submit the data is sent to the API and user is routed to next step
- When revisiting the POC page, contact info from the Filing API is pre-populated in the form
- On second submit, when no data has changed, no API call is made

https://github.com/cfpb/sbl-frontend/assets/2592907/d7e7b851-ef40-4600-ae57-13adcbc655b0

### CODE REVIEW

File: src/pages/PointOfContact/index.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable react/require-default-props */
```
Comment (billhimmelsbach) [inline src/pages/PointOfContact/index.tsx:1]: Want to just disable this rule for now?

See: https://github.com/cfpb/sbl-frontend/issues/355

File: src/pages/PointOfContact/index.tsx [hunk 2]
```diff
@@ -157,6 +200,7 @@ function PointOfContact(): JSX.Element {
                     { label: '-- select an option --', value: '' },
                     ...statesObject.states,
                   ]}
+                  value={watch('hq_address_state')}
```
Comment (billhimmelsbach) [inline src/pages/PointOfContact/index.tsx:203]: Solid change, thanks for fixing the behavior here.

File: src/pages/PointOfContact/index.tsx [hunk 3]
```diff
@@ -65,12 +99,21 @@ function PointOfContact(): JSX.Element {
...
+            data: formattedUserProfileObject,
+            lei,
+            filingPeriod: year,
+          });
+        }
+
+        if (onSubmit) onSubmit();
       } catch (error) {
         // eslint-disable-next-line no-console
         console.log(error);
```
Comment (billhimmelsbach) [inline src/pages/PointOfContact/index.tsx:119]: Some kind of error box for a later PR right?

**Review by billhimmelsbach — APPROVED** [review]
Looks good! I had a few comments that you can scoop up for a second PR if you'd like.

Also, one more thing for the next PR would be to show a loading state before the API call completes, instead of showing a blank form that fills in with data. Slow internet connections would make this experience a little weird.

**Comment by billhimmelsbach** [comment]
> Also, one more thing for the next PR would be to show a loading state before the API call completes, instead of showing a blank form that fills in with data. Slow internet connections would make this experience a little weird.

It's Bill from the Future who just reviewed your next PR, which has those inputs disabled on load, which I think will definitely work for MVP. I do think natalia will have thoughts about the various load states later though, so can we make a post-mvp ticket to re-visit form loading states?

**Comment by meissadia** [comment]
> > Also, one more thing for the next PR would be to show a loading state before the API call completes, instead of showing a blank form that fills in with data. Slow internet connections would make this experience a little weird.
> 
> It's Bill from the Future who just reviewed your next PR, which has those inputs disabled on load, which I think will definitely work for MVP. I do think natalia will have thoughts about the various load states later though, so can we make a post-mvp ticket to re-visit form loading states?

https://github.com/cfpb/sbl-frontend/issues/459

File: src/pages/PointOfContact/index.tsx (hunk 1 above)
Comment (meissadia) [reply src/pages/PointOfContact/index.tsx:1]: @billhimmelsbach Pulled that ticket into the Sprint, but will probably make the change next sprint.  And will remove all of these `disable`s as part of that work.

File: src/pages/PointOfContact/index.tsx (hunk 3 above)
Comment (meissadia) [reply src/pages/PointOfContact/index.tsx:119]: @billhimmelsbach Yes

**Comment by meissadia** [comment]
Added todo items in [POC - Phase 2](https://github.com/cfpb/sbl-frontend/issues/451) ticket
- Full page loading state instead of form level
- Error alert for API submission errors

**Pull request merged**